plugins/cilium-cni/main.go

```diff
@@ -552,6 +557,9 @@ func cmdAdd(args *skel.CmdArgs) (err error) {
 			err = fmt.Errorf("unable to prepare IP addressing for '%s': %s", ep.Addressing.IPV6, err)
 			return
 		}
+		// Add to the result the Interface as index of Interfaces
+		ipConfig.Interface = cniTypesV1.Int(len(res.Interfaces))
```

Comment:
Shouldn't this be:
```suggestion
		ipConfig.Interface = cniTypesV1.Int(len(res.Interfaces)-1)
```

Reply:
Sorry I'm still confused,

On line: https://github.com/cilium/cilium/pull/24901/files#diff-87793c7d9c6d6a10f34d6a3efefed53dac8abd700cb05cfb2a77844ec990b744R523

So at this point res.Interfaces is either of length 0 or 1, if ip4 and ip6 are enabled then we're going to be referencing 2, which is out of bounds in the Interfaces array?

Does that make sense? This function is a bit confusing so I might be missing something here.

Reply:
On line L523, we add the host-side veth info to `res.Interface` which has no ip.
On line L561 and L577, `ipConfig` was refer to container-side veth info. `ipConfig` is the **last element** that will be added to `res.Interface` slice. So I think
```go
// ipConfig.Interface represents its index in this slice.
ipConfig.Interface = cniTypesV1.Int(len(res.Interfaces))
```
is what we want.

Reply:
Does that still work if datapathmodeveth isn't enabled?

Reply:
Yes. We add host side veth info to `res.Interfaces` only when datapathmodeveth enabled. Everything is the same as before if datapathmodeveth was not enabled.

Reply:
Ok I see now, thanks for the clarification.